ast/parser/EarthParser.g4

```diff
@@ -141,5 +141,5 @@ importStmt: IMPORT (WS stmtWords)?;
 expr: stmtWordsMaybeJSON;
 
 stmtWordsMaybeJSON: stmtWords;
-stmtWords: stmtWord (WS? stmtWord)*;
+stmtWords: stmtWord (WS stmtWord)*;
```

Comment:
I wonder if we can ever have `stmtWords` without whitespace. e.g.

```earthly
RUN "hello""world"
```

Reply:
Good point.

```Earthfile
FROM ubuntu:latest
RUN echo "hello""world"
```

works with both v0.6.2 and this PR.

I'll try some more test cases to see if I can break it.

Reply:
This works too.
```Earthly
FROM ubuntu:latest
RUN "ec""ho" "hello world"
```

Reply:
and this:
```Earthly
FROM ubuntu:latest
RUN echo \
"this""too"
```

Reply:
Maybe also try `RUN ["echo","hello","world"]`

Reply:
Even this one works:

```Earthly
FROM ubuntu:20.10

a:
    RUN "da""te"

b:
    RUN ["echo","hello","world"]

all:
    BUILD "+""a"
    BUILD "+"\
"b"
```

```
       ubuntu:20.10 | --> Load metadata linux/amd64
/home/alex/junk+base | --> FROM ubuntu:20.10
/home/alex/junk+base | [██████████] 100% resolve docker.io/library/ubuntu:20.10@sha256:a7b08558af07bcccca994b01e1c84f1d14a2156e0099fcf7fcf73f52d082791e
   /home/alex/junk+b | --> RUN echo hello world
   /home/alex/junk+a | --> RUN "da""te"
   /home/alex/junk+b | hello world
   /home/alex/junk+a | Wed Jan  5 21:38:22 UTC 2022
              output | --> exporting outputs
```

@vladaionescu Do you think it's worthwhile adding this one to a test somewhere, perhaps `tests/quotes.earth`?

Reply:
This could simply be an ast test. We could add additional tests like this one: https://github.com/earthly/earthly/blob/main/ast/tests/if.earth in the same directory.

Reply:
I added this as a new test under https://github.com/earthly/earthly/pull/1541 then will rebase this PR against it to show this continues to work.

Reply:
tests are passing after the rebase. I think we're fine.